web/profiles/custom/yalesites_profile/modules/custom/ys_templated_content/src/Modifiers/TemplateModifier.php

```diff
@@ -0,0 +1,228 @@
+   */
+  public function replaceUuids($content, $uuid, $newUuid) {
+    if (array_key_exists('uuid', $content) && $content['uuid'] === $uuid) {
+      $content['uuid'] = $newUuid;
+    }
+    // Find any other element that has the original UUID passed and replace it
+    // with the new one.
+    foreach ($content as $key => $value) {
+      if (is_array($value)) {
+        $content[$key] = $this->replaceUuids($value, $uuid, $newUuid);
+      }
+      elseif (is_string($value) && strpos($value, 'entity:node/') !== FALSE) {
+        $content[$key] = str_replace('entity:node/' . $uuid, 'entity:node/' . $newUuid, $value);
+      }
+      elseif (is_string($value) && strpos($value, $uuid) !== FALSE) {
+        $content[$key] = str_replace($uuid, $newUuid, $value);
+      }
+    }
+
+    return $content;
+  }
+
+  /**
+   * Generate a unique alias using the current date/time.
+   *
+   * @param string $alias
+   *   The alias.
+   *
+   * @return string
+   *   The unique alias (original alias with date/time).
+   */
+  protected function generateUniqueAliasWithDate($alias) {
+    $date = date('Y-m-d-H-i-s');
+    $alias .= '-' . $date;
+
+    return $alias;
+  }
+
+  /**
+   * Generate a unique alias with a sequential number.
+   *
+   * @param string $alias
+   *   The alias.
+   *
+   * @return string
+   *   The unique alias (original alias with sequential number).
+   */
+  protected function generateUniqueAliasWithSequentialNumber($alias) {
+    $aliasNumber = 1;
+    $newAlias = $alias . '-' . $aliasNumber;
+
+    while ($this->pathAliasRepository->lookupByAlias($newAlias, 'en')) {
+      $aliasNumber++;
+      $newAlias = $alias . '-' . $aliasNumber;
+    }
+
+    return $newAlias;
+  }
+
+  /**
+   * Generate a unique alias.
+   *
+   * @param string $alias
+   *   The alias.
+   *
+   * @return string
+   *   The unique alias.
+   */
+  protected function generateUniqueAlias($alias) {
+    return $this->generateUniqueAliasWithSequentialNumber($alias);
+    /* return $this->generateUniqueAliasWithDate($alias); */
+  }
+
+  /**
+   * Generate a unique alias if the alias already exists.
+   *
+   * @param array $content_array
+   *   The content array.
+   *
+   * @return array
+   *   The content array with a unique alias.
+   */
+  protected function generateAlias($content_array) {
+    if (isset($content_array['base_fields']['url'])) {
+      $alias = $content_array['base_fields']['url'];
+
+      if ($this->pathAliasRepository->lookupByAlias($alias, 'en')) {
+        $content_array['base_fields']['url'] = $this->generateUniqueAlias($alias);
+      }
+    }
+
+    return $content_array;
+  }
+
+  /**
+   * Generate a UUID.
+   *
+   * @return string
+   *   The UUID.
+   */
+  public function generateUuid() {
+    return $this->uuidService->generate();
+  }
+
+  /**
+   * Get the current unix timestamp.
+   *
+   * @return int
+   *   The current unix timestamp.
+   */
+  public function getUnixTimestamp() {
```

Comment:
This method feels a little silly to have. I love small methods, but I can't see a case where simply using `time()` isn't just as good. To each their own though.

Reply:
I don't mind swapping it.  HAHAH  I wish I could even tell you why it was split out now that I see it again.  hahah